docker_fargate/docker_fargate_stack.py

```diff
@@ -1,155 +1,142 @@
 
-def get_hosted_zone_name() -> str:
-	return os.getenv(HOSTED_ZONE_NAME)
-	
-def get_hosted_zone_id() -> str:
-	return os.getenv(HOSTED_ZONE_ID)
-	
-def get_host_name() -> str:
-	return os.getenv(HOST_NAME)
 
 class DockerFargateStack(Stack):
 
-    def __init__(self, scope: Construct, **kwargs) -> None:
-        stack_id = create_id()
+    def __init__(self, scope: Construct, context: str, env: dict, vpc: ec2.Vpc, **kwargs) -> None:
+        stack_id = create_id(env)
         super().__init__(scope, stack_id, **kwargs)
 
-        vpc = ec2.Vpc(self, get_vpc_name(), max_azs=2)
-
-        cluster = ecs.Cluster(self, get_cluster_name(), vpc=vpc, container_insights=True)
+        cluster = ecs.Cluster(self, get_cluster_name(env), vpc=vpc, container_insights=True)
 
+        secret_name = f'{stack_id}/{context}/ecs'
         secrets = {
-        	SECRETS_MANAGER_ENV_NAME: create_secret(self, get_secret_name())
+            SECRETS_MANAGER_ENV_NAME: get_secret(self, secret_name, secret_name)
         }
 
         env_vars = {}
-        container_env = get_container_env()
+        container_env = get_container_env(env)
         if container_env is not None:
             env_vars[ENV_NAME]=container_env
 
         task_image_options = ecs_patterns.ApplicationLoadBalancedTaskImageOptions(
-    	    	   image=ecs.ContainerImage.from_registry(get_docker_image_name()),
-    	    	   environment=env_vars,
-    	    	   secrets = secrets,
-    	    	   container_port = get_port())
-        
-        zone = r53.PublicHostedZone.from_public_hosted_zone_attributes(self, id=stack_id+"_zone", hosted_zone_id=get_hosted_zone_id(), zone_name=get_hosted_zone_name())
-
- 
+                   image=ecs.ContainerImage.from_registry(get_docker_image_name(env)),
+                   environment=env_vars,
+                   secrets = secrets,
+                   container_port = get_port(env))
+
+        zone = r53.PublicHostedZone.from_public_hosted_zone_attributes(
+            self,
+            id=stack_id+"_zone",
+            hosted_zone_id=get_hosted_zone_id(env),
+            zone_name=get_hosted_zone_name(env))
+
+
         #
         # for options to pass to ApplicationLoadBalancedTaskImageOptions see:
         # https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_ecs_patterns/ApplicationLoadBalancedTaskImageOptions.html#aws_cdk.aws_ecs_patterns.ApplicationLoadBalancedTaskImageOptions
         #
         load_balanced_fargate_service = ecs_patterns.\
-                ApplicationLoadBalancedFargateService(self, get_service_name(),
+                ApplicationLoadBalancedFargateService(self, get_service_name(env),
             cluster=cluster,            # Required
             cpu=256,                    # Default is 256
-            desired_count=3,            # Number of copies of the 'task' (i.e. the app') running behind the ALB
+            desired_count=1,            # Number of copies of the 'task' (i.e. the app') running behind the ALB
```

Comment:
it looks like this value was intentionally increased in commit b60d4f1, we should keep it at `3`

Reply:
@zaro0508 For quite some time, we were discussing about `server unavailable error` [here](https://sagebionetworks.jira.com/browse/DCA-51), and increasing the desired_count [here](https://sagebionetworks.jira.com/browse/DCA-44)

I noticed that if I enabled multiple app running, the server could come back up faster when the error came out. But since we now switch to use uWSGI, we might be able to switch it back to 1 and it will work the same. (But I am not sure!) Since we are still trouble shooting gate-way time out error, can we keep this as 3? After we completely figure  out the error, we could experiment again to reduce this back to 1. I want to keep this variable the same as I do testing so that I don't have to consider another factor.

Reply:
sounds good.  i've change updated it to 3.